identity_credential/src/credential/proof.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020-2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use identity_core::common::Object;
+use serde::Deserialize;
+use serde::Serialize;
+
+/// Represents a cryptographic proof that can be used to validate verifiable credentials and
+/// presentations.
+///
+/// This representation does not inherently implement any standard; instead, it
+/// can be utilized to implement standards or user-defined proofs. The presence of the
+/// `type` field is necessary to accommodate different types of cryptographic proofs.
+///
+/// Note that this proof is not related to JWT and can be used in combination or as an alternative
+/// to it.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct Proof {
+  /// Type of proof.
+  #[serde(rename = "type")]
+  pub _type: String,
+
+  /// Properties of the proof, entries depend on the proof type.
+  #[serde(flatten)]
+  pub properties: Object,
+}
+
+impl Proof {
+  /// Creates a new `Proof`.
+  pub fn new(_type: String, properties: Object) -> Proof {
+    Self { _type, properties }
+  }
+}
+
+#[cfg(test)]
+mod tests {
+  use super::Proof;
+  use crate::credential::Credential;
+  use identity_core::common::Object;
+  use identity_core::convert::FromJson;
+
+  #[test]
+  fn test_from_json() {
+    let proof = r#"
+    {
+      "type": "test-proof",
+      "signature": "abc123"
+    }
+    "#;
+    let proof: Proof = Proof::from_json(proof).expect("proof deserialization failed");
+
+    assert_eq!(proof._type, "test-proof");
+    let value = proof
+      .properties
+      .get(&"signature".to_owned())
+      .expect("property in proof doesn't exist");
+    assert_eq!(value, "abc123");
+  }
+
+  #[test]
+  fn test_credential_from_json() {
```

Comment:
```suggestion
  fn test_credential_with_proof_from_json() {
```

Nit: Be more specific in what this test really tests.